plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/errors/VerificationError.kt

```diff
@@ -45,4 +47,13 @@ object ErrorAdapter {
             else -> UnknownError(err)
         }
     }
-}
\ No newline at end of file
+}
+
+/**
+ * Given a verification error, find embedded extra information of type `I` in the
+ * error's offending nodes. If such information is not found, return `default` as value.
+ */
```

Comment:
This seems to be neither a full tree search nor just an immediate lookup.  Could you document what it is looking for?  Are calls like `_2()` always safe to do, or could this result in some kind of out-of-bounds exception?

Reply:
> This seems to be neither a full tree search nor just an immediate lookup. Could you document what it is looking for?

Sure! I will write a short summary.

> Are calls like _2() always safe to do, or could this result in some kind of out-of-bounds exception?

Yes, they are calls to access Scala Tuples`N`'s elements (in this case `Tuple3` exposes three accessor methods `_1, _2, _3`).